internal/provisioner/registry/README.md

```diff
@@ -1,3 +1,278 @@
-# registry
+# Registry Provisioner
 
-This provisioner was removed in commit 76fefcfcd1408d04b4eb7eb368d3eaab49565237.
+## Summary
+
+The `registry` provisioner is a core Rukpak provisioner that knows how to interact with bundles of a particular format.
+These `registry+v1` bundles, or `registry` bundles, are container images containing a set of static Kubernetes YAML
+manifests organized in the legacy Operator Lifecycle Manger (OLM) format. For more information on the `registry+v1` format, see
+the [OLM packaging doc](https://docs.openshift.com/container-platform/4.10/operators/understanding/olm-packaging-format.html).
+
+The `registry` provisioner is able to convert a given `registry+v1` bundle onto a cluster in the `plain+v0` format. Instantiation of the
+bundle is then handled by the `plain` provisioner in order to make the content available in the cluster. It does so by reconciling `Bundle` 
+types that have the `spec.provisionerClassName` field set to `core.rukpak.io/registry`. This field must be set to the correct provisioner 
+name in order for the `registry` provisioner to see and interact with the bundle. Creating a `BundleInstance` of that `Bundle` would then 
+require that you have set the `BundleInstance` property `spec.provisionerClassName` field to `core.rukpak.io/plain`.
+
```

Comment:
Might be helpful to drop an example BI spec right here? The explanation above is hard to visualize. Or maybe we just could add a diagram of some kind?

Reply:
I put a link to the section of the document where we start defining this since it felt alike we could reuse some text here.